mlx/backend/metal/reduce.h

```diff
@@ -37,6 +37,7 @@ void strided_reduce_general_dispatch(
     const std::vector<int>& axes,
     CommandEncoder& compute_encoder,
     metal::Device& d,
-    const Stream& s);
+    const Stream& s,
+    std::vector<array>& copies);
```

Comment:
Just as a side note, this idiom is not so critical anymore (though I certainly don't mind if we keep it this way). I think (?) we started writing functions with temporaries pass-by-reference vectors to avoid completion handler bloat. But if you do a `d.add_temporaries` inside the `strided_reduce...` it will put the temporary buffer in a vector and we only add one completion handler at the very end after the command buffer is committed.

Reply:
Oh you 're right! Force of habit. I will change all the reductions to use that idiom because it makes the internal API simpler. (see normalization implementations in `fast` for instance).